activity_calendar/templatetags/calendar_tags.py

```diff
@@ -50,12 +50,13 @@ def opens_in_future(context):
 @register.inclusion_tag("activity_calendar/slot_blocks/register_button.html", takes_context=True)
 def register_button(context, slot):
     user = context['request'].user
+    participants = context['participants']
 
     # There is no logged in user, so return nothing instead
     if not user.is_authenticated:
         return None
 
-    sign_up = not slot.participants.filter(id=user.id).exists()
+    sign_up = user not in participants
```

Comment:
To my knowledge the not in function uses python selection whereas the original query does searching on the database.
While current implementation does indeed save a database query (why this is an issue is beyond me) it does require participants to be in the local context when using this tag. That does not make sense (note that any call to context I made when writing these tags are guarenteed as they are covered in test cases). If you really want to save the database query this way, please instead require this tag to have participants as an input argument instead.